sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala

```diff
@@ -96,6 +96,8 @@ private[sql] class TypeCollection(private val types: Seq[AbstractDataType])
 
 private[sql] object TypeCollection {
 
+  val Ordered = TypeCollection(NumericType, StringType)
```

Comment:
the thing is i'm not sure if atomictype is the right one to enforce ordering.

e.g. I think we should be able to sort by struct or array too.